### PR TITLE
Jira fixes

### PR DESCRIPTION
**Pull request by jspricke**

### CODE REVIEW

File: bugwarrior/services/jira.py [hunk 1]
```diff
@@ -188,8 +188,11 @@ def __get_sprints(self):
             for key in self.origin['sprint_field_names']
         ], [])
         for sprint in sprints:
-            # Parse this big ugly string.
-            yield _parse_sprint_string(sprint)
+            if isinstance(sprint, dict):
+                yield sprint
+            else:
```
Comment (ryneeverett) [inline bugwarrior/services/jira.py:193]: I'd like to add a comment here like `else:  # Old version of python-jira without sprint support.`

**Review by ryneeverett — APPROVED** [review]

File: bugwarrior/services/jira.py (hunk 1 above)
Comment (jspricke) [reply bugwarrior/services/jira.py:193]: This is actually not a new feature of python-jira but Atlassian seems to have fixed it in Jira itself. I left the old code as a backup as I don't know if it was fixed everywhere.

Comment (ryneeverett) [reply bugwarrior/services/jira.py:193]: I see. I still think some comment indicating that the else-clause is for backwards compatibility would be nice.

Comment (jspricke) [reply bugwarrior/services/jira.py:193]: Agreed, I added a comment.

**Review by ryneeverett — APPROVED** [review]

File: bugwarrior/services/jira.py [hunk 2]
```diff
@@ -159,7 +159,7 @@ def get_due(self):
             return self.parse_date(self.record['fields']['duedate'])
         # Otherwise, if the issue is in a sprint, use the end date of that sprint.
         sprints = self.__get_sprints()
-        for sprint in filter(lambda e: e.get('state') != 'CLOSED', sprints):
+        for sprint in filter(lambda e: e.get('state', '').lower() != 'closed', sprints):
```
Comment (StephanMeijer) [inline bugwarrior/services/jira.py:162]: Can we make a constant of `closed`?

**Review by StephanMeijer — CHANGES_REQUESTED** [review]

File: bugwarrior/services/jira.py (hunk 2 above)
Comment (jspricke) [reply bugwarrior/services/jira.py:162]: The string is generated by Jira and exported through it's API. python-jira has nothing about it and this is the only place it is used in the bugwarrior jira.py. Why would you want to make it a constant?

**Pull request merged**